kong/plugins/grpc-gateway/handler.lua

```diff
@@ -50,13 +48,14 @@ function grpc_gateway:access(conf)
 
   kong.ctx.plugin.dec = dec
 
-  kong_service_request_set_header("Content-Type", "application/grpc")
-  kong_service_request_set_header("TE", "trailers")
   local body, err = dec:upstream(kong_request_get_raw_body())
   if err then
     kong.log.err(err)
-    return kong_response_exit(400, err)
+    return kong_response_exit(400, err, { ["Content-Type"] = "text/html" })
```

Comment:
Good catch. The error response is not of the grpc MIME type.

We usually do not manually set CT for the error response.

Reply:
Can it stay with this CT?

Reply:
It should be OK I guess.

Reply:
Is this one resolved?